storage/innobase/log/log0recv.cc

```diff
@@ -762,14 +762,15 @@ void recv_sys_t::open_log_files_if_needed()
   }
 }
 
-void recv_sys_t::read(os_offset_t total_offset, span<byte> buf)
+MY_ATTRIBUTE((warn_unused_result))
+dberr_t
+recv_sys_t::read(os_offset_t total_offset, span<byte> buf)
 {
   open_log_files_if_needed();
 
   size_t file_idx= static_cast<size_t>(total_offset / log_sys.log.file_size);
   os_offset_t offset= total_offset % log_sys.log.file_size;
-  dberr_t err= recv_sys.files[file_idx].read(offset, buf);
-  ut_a(err == DB_SUCCESS);
+  return recv_sys.files[file_idx].read(offset, buf);
```

Comment:
While you are at this, you might want to include parts of c72ddeea4f5400f02b65df19c2e2c02da8166c1f, to make sure that the first log file is only registered in one place, and that `recv_sys.close_files()` will actually close the files.

Reply:
Sorry, I did not realize that I had refactored `log_file_t` in 685d958e38b825ad9829be311f26729cccf37c46. Please disregard my previous comment.